lib/popcorn/pop_handlers.nit

```diff
@@ -0,0 +1,427 @@
+
+	# POST handler.
+	#
+	# Exemple of route responding to POST requests.
+	# ~~~
+	# class PostHandler
+	#	super Handler
+	#
+	#	redef fun post(req, res) do res.send "POST request received"
+	# end
+	# ~~~
+	fun post(req: HttpRequest, res: HttpResponse) do end
+
+	# PUT handler.
+	#
+	# Exemple of route responding to PUT requests.
+	# ~~~
+	# class PutHandler
+	#	super Handler
+	#
+	#	redef fun put(req, res) do res.send "PUT request received"
+	# end
+	# ~~~
+	fun put(req: HttpRequest, res: HttpResponse) do end
+
+	# DELETE handler.
+	#
+	# Exemple of route responding to PUT requests.
+	# ~~~
+	# class DeleteHandler
+	#	super Handler
+	#
+	#	redef fun delete(req, res) do res.send "DELETE request received"
+	# end
+	# ~~~
+	fun delete(req: HttpRequest, res: HttpResponse) do end
+end
+
+# Static files server.
+#
+# To serve static files such as images, CSS files, and JavaScript files, use the
+# Popcorn built-in handler `StaticHandler`.
+#
+# Pass the name of the directory that contains the static assets to the StaticHandler
+# init method to start serving the files directly.
+# For example, use the following code to serve images, CSS files, and JavaScript files
+# in a directory named `public`:
+#
+# ~~~
+# var app = new App
+# app.use("/", new StaticHandler("public/"))
+# ~~~
+#
+# Now, you can load the files that are in the `public` directory:
+#
+# ~~~raw
+# http://localhost:3000/images/trollface.jpg
+# http://localhost:3000/css/style.css
+# http://localhost:3000/js/app.js
+# http://localhost:3000/hello.html
+# ~~~
+#
+# Popcorn looks up the files relative to the static directory, so the name of the
+# static directory is not part of the URL.
+# To use multiple static assets directories, add the `StaticHandler` multiple times:
+#
+# ~~~
+# app.use("/", new StaticHandler("public/"))
+# app.use("/", new StaticHandler("files/"))
+# ~~~
+#
+# Popcorn looks up the files in the order in which you set the static directories
+# with the `use` method.
+#
+# To create a virtual path prefix (where the path does not actually exist in the file system)
+# for files that are served by the `StaticHandler`, specify a mount path for the
+# static directory, as shown below:
+#
+# ~~~
+# app.use("/static/", new StaticHandler("public/"))
+# ~~~
+#
+# Now, you can load the files that are in the public directory from the `/static`
+# path prefix.
+#
+# ~~~raw
+# http://localhost:3000/static/images/trollface.jpg
+# http://localhost:3000/static/css/style.css
+# http://localhost:3000/static/js/app.js
+# http://localhost:3000/static/hello.html
+# ~~~
+#
+# However, the path that you provide to the `StaticHandler` is relative to the
+# directory from where you launch your app.
+# If you run the app from another directory, it’s safer to use the absolute path of
+# the directory that you want to serve.
+class StaticHandler
+	super Handler
+
+	# Static files directory to serve.
+	var static_dir: String
+
+	private var file_server: FileServer is lazy do
```

Comment:
why not exposing the FileServe? so people can tweak it, eg.g reactivating the directory listing